block_iagora.php

```diff
@@ -47,35 +47,43 @@ public function get_content() {
         }
 
         $this->content = new stdClass();
-        $this->content->footer = '';
-
         $iframeurl = isset($this->config->iframeurl) ? $this->config->iframeurl : '';
-
         if (empty($iframeurl)) {
             $this->content->text = get_string('noiframeurl', 'block_iagora');
         } else {
             $this->content->text = '<iframe src="' . $iframeurl . '" width="100%" height="400px" frameborder="0"></iframe>';
         }
-
+        $this->content->text = $this->generate_chat_content($iframeurl);
         return $this->content;
     }
-
     /**
-     * Define in which pages this block can be added.
+     * Generate the HTML content for the chat block.
      *
-     * @return array of the pages where the block can be added.
+     * @param string $iframeurl The URL to be used in the iframe.
+     * @return string The generated HTML content for the chat.
      */
-    public function applicable_formats() {
-        return ['all' => true];
+    private function generate_chat_content($iframeurl) {
+        global $PAGE, $OUTPUT;
+        // Generate a unique identifier for the chat container
+        $chat_id = uniqid('iagora_chat_');
```

Comment:
```suggestion
        $chatid = uniqid('iagora_chat_');
```
See [moodle variables naming convention](https://moodledev.io/general/development/policies/codingstyle#variables).